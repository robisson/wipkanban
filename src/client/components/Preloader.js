// @flow
import * as React from "react";
import { withStyles } from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";

type Props = {
  classes: Object,
  showPreloader: Boolean
};

const styles = () => ({
  background: {
    position: "absolute",
    width: "100%",
    height: "100vh",
    backgroundColor: "black",
    opacity: 0.5
  },
  wrapper: {
    position: "absolute",
    width: "100%",
    height: "100vh",
    top: 0,
    left: 0,
    zIndex: 10000,
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  }
});

const Preloader = ({ classes, showPreloader }: Props) => {
  if (!showPreloader) {
    return "";
  }

  return (
    <div className={classes.wrapper}>
      <div className={classes.background} />
      <CircularProgress
        className={classes.progress}
        color="secondary"
        size={50}
      />
    </div>
  );
};

export default withStyles(styles)(Preloader);
