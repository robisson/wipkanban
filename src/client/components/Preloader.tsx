import * as React from "react";
import { withStyles } from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";

interface IPreloaderProps {
  classes: any;
  showPreloader: Boolean;
}

const styles: any = () => ({
  background: {
    position: "absolute",
    width: "100%",
    height: "100vh",
    backgroundColor: "black",
    opacity: 0.5
  },
  wrapper: {
    position: "absolute",
    width: "100%",
    height: "100vh",
    top: 0,
    left: 0,
    zIndex: 10000,
    display: "flex",
    justifyContent: "center",
    alignItems: "center"
  }
});

const Preloader: React.StatelessComponent<any> = ({
  classes,
  showPreloader
}: IPreloaderProps) => {
  if (!showPreloader) {
    return null;
  }

  return (
    <div className={classes.wrapper}>
      <div className={classes.background} />
      <CircularProgress
        className={classes.progress}
        color="secondary"
        size={50}
      />
    </div>
  );
};

export default withStyles(styles)(Preloader);
