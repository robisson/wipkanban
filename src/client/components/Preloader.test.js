import React from "react";
import toJson from "enzyme-to-json";
import { createShallow } from "@material-ui/core/test-utils";
import "../setupTest";
import Preloader from "./Preloader";
import CircularProgress from "@material-ui/core/CircularProgress";

describe("<Preloader /> Component", () => {
  let wrapper;
  let shallow;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it("should return empty content when showPrealoder is not defined", () => {
    wrapper = shallow(<Preloader />);
    expect(wrapper.type()).toBeNull;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should to render <CircularProgress /> when showPreloader is true", () => {
    wrapper = shallow(<Preloader showPreloader={true} />);
    expect(wrapper.find(CircularProgress).type()).toEqual(CircularProgress);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
