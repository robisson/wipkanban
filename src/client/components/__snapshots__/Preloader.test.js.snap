// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader /> Component should return empty content when showPrealoder is not defined 1`] = `""`;

exports[`<Preloader /> Component should to render <CircularProgress /> when showPreloader is true 1`] = `
<div
  className="Preloader-wrapper-2"
>
  <div
    className="Preloader-background-1"
  />
  <WithStyles(CircularProgress)
    color="secondary"
    size={50}
  />
</div>
`;
